source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/inputhandler/RatingInputHandler.kt

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package io.adaptivecards.renderer.inputhandler
+
+import android.view.accessibility.AccessibilityEvent
+import io.adaptivecards.objectmodel.BaseInputElement
+import io.adaptivecards.objectmodel.RatingInput
+import io.adaptivecards.renderer.Util
+import io.adaptivecards.renderer.layout.RatingStarInputView
+import io.adaptivecards.renderer.layout.RatingStarInputViewListener
+
+/**
+ * Input handler for Rating input
+ **/
+class RatingInputHandler(
+    baseInputElement: BaseInputElement
+): BaseInputHandler(baseInputElement) {
+    override fun getInput() = (m_view as RatingStarInputView).rating.toString()
+
+    override fun setInput(input: String) {
+        try {
+            (m_view as RatingStarInputView).handleRatingChange(input.toDouble())
+        } catch(e: NumberFormatException) {
+            return
+        }
+    }
+
+    override fun setFocusToView() {
+        val focusView = (m_view as RatingStarInputView).getChildAt(0)
+        Util.forceFocus(focusView)
+        focusView.sendAccessibilityEvent(AccessibilityEvent.TYPE_VIEW_ACCESSIBILITY_FOCUSED)
+    }
+
+    override fun isValid(showError: Boolean): Boolean {
+        var isValid = true
+        if (m_baseInputElement.GetIsRequired()) {
+            isValid = try {
+                input.toDouble() > 0
+            } catch(e: NumberFormatException) {
+                false
+            }
+        }
+        isValid = isValid && isValidOnSpecifics(input)
+        if (showError) {
+            showValidationErrors(isValid)
+        }
+        return isValid
+    }
+
+    override fun resetValue() {
+        val ratingInput = Util.castTo(m_baseInputElement, RatingInput::class.java)
+        input = ratingInput.GetValue().toString()
```

Comment:
call setInput(ratingInput.GetValue()) to set the value and change the UI.

Reply:
It's setting the value only. Used Kotlin property access syntax.
UI is getting updated while setting the rating. Refer setInput method which calls handleRatingChange method to set the rating and updates the view.